posthog/models/kafka_engine_dlq/sql.py

```diff
@@ -0,0 +1,36 @@
+# Every table using the Kafka Engine should have this corresponding
+# dead letter queue (DLQ) table. This table will host all messages
+# that couldn't be consumed by the ClickHouse consumer for the topic
+# and inserted into the desired table and their corresponding errors.
+# This can then be used to set up monitoring and debugging issues
+# with the messages in Kafka, either because we've produced invalid
+# messages (which we shouldn't), or because ClickHouse couldn't
+# consume them due to a bug or otherwise. This is essential
+# to make sure we're not silently dropping events and so we
+# can have full oversight over the end-to-end ingestion pipeline.
+
+KAFKA_ENGINE_DLQ_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table} ON CLUSTER '{cluster}'
```

Comment:
Could we get _timestamp in here as well and partition by it?

We will likely want to partition by that as well and perhaps provide options to drop old DLQ entries. Currently adding TTLs would involve rewriting ALL_OF_DATA.

Daily plug of the wip-manual: https://posthog.com/handbook/engineering/clickhouse/data-storage#partition-by